app/src/main/kotlin/compiler/lexer/dfa/RegexDfa.kt

```diff
@@ -1,23 +1,22 @@
 package compiler.lexer.dfa
 
 import compiler.lexer.regex.Regex
+import compiler.lexer.regex.RegexFactory
 
-class RegexDfa(var regex: Regex) : Dfa {
-
+class RegexDfa(private val regex: Regex) : Dfa {
     override fun newWalk(): DfaWalk {
+        var currentStateRegex = regex
```

Comment:
It doesn't matter, because in Kotlin you can capture all variables from outer scope https://stackoverflow.com/questions/44538627/why-the-code-in-an-object-expression-can-access-variables-from-the-scope-that-co, but perhaps it would be a little bit more readable if this var would be inside of the returned object.